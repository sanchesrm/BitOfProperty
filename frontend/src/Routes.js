import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import Login from "./Login";
import mainPage from "./mainPage";
import { Layout } from "antd";
import "antd/dist/antd.css";

const { Header, Content } = Layout;

const Routes = () => (
  <BrowserRouter>
    <main>
      <Route path="/login" exact component={Login} />
      <Header>
        <Content>
          <Route path="/" component={mainPage} />
        </Content>
      </Header>
    </main>
  </BrowserRouter>
);

export default Routes;
