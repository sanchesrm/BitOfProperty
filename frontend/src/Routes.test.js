import Adapter from "enzyme-adapter-react-16";
import React from "react";
import toJson from "enzyme-to-json";
import { configure, shallow } from "enzyme";
import { Routes } from "./Routes";

describe("Routes Component", () => {
  let props;
  beforeEach(() => {
    configure({ adapter: new Adapter() });
    props = { loggedUser: [] };
  });

  test("render correctly", () => {
    expect.assertions(1);

    const wrapper = shallow(<Routes {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("render correctly when token", () => {
    expect.assertions(1);

    props = {
      ...props,
      loggedUser: {
        token: "1234"
      }
    };

    const wrapper = shallow(<Routes {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
