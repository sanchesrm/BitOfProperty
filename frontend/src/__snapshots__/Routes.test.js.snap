// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component render correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Redirect
      push={false}
      to="/login"
    />
  </Switch>
</BrowserRouter>
`;

exports[`Routes Component render correctly when token 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <div>
      <Adapter>
        <img
          alt=""
          src="bitofproperty.png"
          style={
            Object {
              "float": "left",
              "marginLeft": "150px",
              "marginTop": "2px",
              "width": "270px",
            }
          }
        />
        <Menu
          className=""
          focusable={false}
          mode="horizontal"
          prefixCls="ant-menu"
          style={
            Object {
              "lineHeight": "64px",
            }
          }
          theme="dark"
        >
          <MenuItem
            key="1"
            onClick={[Function]}
            style={
              Object {
                "float": "right",
                "fontSize": "18px",
                "marginRight": "150px",
              }
            }
          >
            Logout
          </MenuItem>
        </Menu>
      </Adapter>
      <Adapter
        style={
          Object {
            "padding": "50px 20%",
          }
        }
      >
        <Switch>
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>
      </Adapter>
    </div>
  </Switch>
</BrowserRouter>
`;
